packages/react-components/src/types/OnRender.ts

```diff
@@ -61,4 +61,4 @@ export type OnRenderAvatarCallback = (
    * Pass the `options` to the `onRender` component for default rendering.
    */
   defaultOnRender?: (props: CustomAvatarOptions) => JSX.Element
-) => JSX.Element;
+) => JSX.Element | undefined;
```

Comment:
Note, this is not a breaking change as it is now _less_ restrictive and existing code will function fine.